tcconfig/shaper/htb.py

```diff
@@ -15,6 +15,26 @@
 from .._network import get_upper_limit_rate
 from .._tc_command_helper import run_tc_show
 from ._interface import AbstractShaper
+from pyroute2.netlink.rtnl.tcmsg.common import  (
+    get_hz,
+    tick_in_usec,
+    TIME_UNITS_PER_SEC
+)
+
+# Emulation of tc's buggy time2tick implementation, which
+# rounds to int twice
+def time2tick_bug(time):
+    return int(int(time) * tick_in_usec)
+
+# An accurate implementation of tick2time (unlike tc's), not rounding.
+def tick2time_true(tick):
+    return tick / tick_in_usec
+
+def calc_xmittime_bug(rate, size):
+    return int(time2tick_bug(TIME_UNITS_PER_SEC * (float(size) / rate)))
+
+def calc_xmitsize_true(rate, ticks):
+    return (float(rate)*tick2time_true(ticks))/TIME_UNITS_PER_SEC
```

Comment:
Could you add some simple unit tests to these functions?

Reply:
The difficulty here is that `tick_in_usec` is a system-dependent value, so it's hard to write reproducible tests for these functions.

I could override it in the tests, but that's scribbling on the inside of pyroute2's internal state, which seems fraught at best?

Reply:
Alternately, I could add an extra parameter to the function to allow me to pass in an external value of `tick_in_usec` for testing, but then the tests don't use the same code path as the production code...

Reply:
How about adding a wrapper function that returns `tick_in_usec` like the following instead of using `tick_in_usec` directly:

```py
def get_tick_in_usec() -> int:
    return tick_in_usec
```

This way mocks will not affect anything inside `pyroute2`

Reply:
Done.